collect_reprogs.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import filecmp
+import logging
+import os
+import shutil
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Collect C reproducers programs from subdirs.')
+    parser.add_argument('directory', type=str,
+                        help='Folder to be searched for reprogs.')
+    parser.add_argument('--output', '-o', required=False, default="./linux",
+                        help='Output folder where to store. default ./linux')
+    parser.add_argument('--collect_all', '-c', required=False, default=False, action='store_true',
+                        help='There might be more than one C reproducers per bug. Use this if you want to collect them all. \
+                            By default only the first one is added to the output dir.')
+    args = parser.parse_args()
+
+    workdir = os.path.join(args.directory, "")
+    logging.debug(workdir)
+
+    for file in glob.iglob(workdir + '**/*.cprog', recursive=True):
+        logging.debug(file)
+        file_path = file.split('/')
+        has_id_index = len(file_path) - 2
+
+        candidate_name = file_path[has_id_index]
+        candidate_file_name = os.path.join(args.output, candidate_name + ".c")
+        index = 1
+        already_exist = False
+        while os.path.exists(candidate_file_name):
+            if filecmp.cmp(candidate_file_name, file, shallow=False):
+                already_exist = True
+                break
+            candidate_file_name = os.path.join(args.output, candidate_name + "_" + str(index).rjust(3, '0') + ".c")
```

Comment:
If I am not mistaken, this still tries to append index. We decided to drop this part.

Reply:
Yes and no. Default action is that it will not save the reproducer with the new name if it already exists.
But if you give the --collect_all -option then. It will save all the different versions.

If you don't like this option I can remove it.

Reply:
I just removed it.